lib/amazonka/src/Amazonka/Send.hs

```diff
@@ -0,0 +1,156 @@
+module Amazonka.Send
+  ( send,
+    sendEither,
+    paginate,
+    paginateEither,
+    await,
+    awaitEither,
+    sendUnsigned,
+    sendUnsignedEither,
+  )
+where
+
+import Amazonka.Env (Env, Env')
+import Amazonka.Core (AWSPager, AWSRequest, AWSResponse, Error)
+import qualified Amazonka.HTTP as HTTP
+import qualified Amazonka.Pager as Pager
+import Amazonka.Prelude
+import qualified Amazonka.Waiter as Waiter
+import qualified Control.Exception as Exception
+import Data.Conduit (ConduitM)
+import qualified Data.Conduit as Conduit
+import qualified Network.HTTP.Client as Client
+
+-- | Send a request, returning the associated response if successful.
+--
+-- See 'send'.
+sendEither ::
+  ( MonadResource m,
+    AWSRequest a
+  ) =>
+  Env ->
+  a ->
+  m (Either Error (AWSResponse a))
+sendEither env =
+  fmap (second Client.responseBody) . HTTP.retryRequest env
+
+-- | Send a request, returning the associated response if successful.
+--
+-- Errors are thrown in 'IO'.
+--
+-- See 'sendEither'.
+send ::
+  ( MonadResource m,
+    AWSRequest a
+  ) =>
+  Env ->
+  a ->
+  m (AWSResponse a)
+send env =
+  sendEither env >=> hoistEither
+
+-- | Make a request without signing it. You will almost never need to
+-- do this, but some authentication methods
+-- (e.g. @sts:AssumeRoleWithWebIentity@ and @sso:GetRoleCredentials@)
+-- require you to exchange a token using an unsigned
+-- request. Amazonka's support for these authentication methods calls
+-- 'sendUnsigned', and we re-export these functions in case you need
+-- to support similar authentication methods in your code.
+--
+-- See 'sendUnsigned'.
+sendUnsignedEither ::
+  ( MonadResource m,
+    AWSRequest a
+  ) =>
+  Env' Proxy ->
+  a ->
+  m (Either Error (AWSResponse a))
+sendUnsignedEither env =
+  fmap (second Client.responseBody) . HTTP.retryRequest env
```

Comment:
I didn't realize these were the same concrete implementation and it's just a type-level restriction. Would it be valuable to DRY?

```hs
sendEither ::
  ( MonadResource m,
    AWSRequest a
  ) =>
  Env ->
  a ->
  m (Either Error (AWSResponse a))
sendEither = sendSomethingEither

sendUnsignedEither ::
  ( MonadResource m,
    AWSRequest a
  ) =>
  Env' Proxy ->
  a ->
  m (Either Error (AWSResponse a))
sendUnsignedEither = sendSomethingEither

sendSomethingEither ::
  ( MonadResource m,
    AWSRequest a
  ) =>
  Env' withAuth ->
  a ->
  m (Either Error (AWSResponse a))
sendSomethingEither env =
  fmap (second Client.responseBody) . HTTP.retryRequest env
```

The expressions pretty simple, so I don't see a huge risk in fixing a bug in one and forgetting the other.

Reply:
I didn't notice either until I put them all next to each other. I think that the giant type signature makes DRYing up the expression pretty unappealing. I don't think these functions have changed or will change that much, so I'm happy to leave as-is.